### ISSUE TITLE
Adding interface ICurveEvaluator<T> to the basecode

### ISSUE THREAD

**Issue by vpenades** [MonoGame/MonoGame#7386]
I would like to request adding this interface to the codebase of monogame types:

```c#
public interface ICurveEvaluator<T>
{        
    T Evaluate(float position);
}
```

This interface would be implemented by the `Curve` type.

Rationale:

Curve works by default with float values, and it is true that with 3 Curve instances it's possible to implement a Vector3 curve of sorts.

But things get more complicated if you want a Quaternion curve, which is something it's used for 3D rotation animation.

Not only that, all sorts of algorythms exist for calculating a curve, linear, quadratic, hermite, etc.... Curves can also be compressed, and have internal dictionaries to make queries faster.

So it's obvious that curves can be implemented in a huge variety of ways, but in the end, the code consuming the curve does not care about the underlaying implementation, it just wants to evaluate the curve at a given point.

Regarding the computational cost of using interfaces (AKA the virtual calls issue), consider this:

In many cases, it's going to be used in the form of `ICurveEvaluator<Vector3>` and `ICurveEvaluator<Quaternion>`, and it's going to be called _once_ for every Vector3 and Quaternion value... which is, I think, less computationally expensive than calling Curve.Evaluate three times to complete a Vector3.

Finally, I haven't considered any particular curve implementation beyond the already existing Curve type. I'm just talking about having the interface for the sake of interoperability.

#### What version of MonoGame does the bug occur on:
- MonoGame 3.8

#### What operating system are you using:
- All

#### What MonoGame platform are you using:

- All

**Comment by Jjagg** [comment]
Seems like a good addition to me! Gives users a standard interface for tweening as well, which is nice for library authors.

**Comment by vpenades** [comment]
pull request just in.

I considered to name it just as ICurve<T> , but then I thought that ICurve<T> might be introduced in the future to define and edit actual curves.... and the evaluator of a curve might not neccesarily be a "curve", it might actually be a proxy or a curve mixer.

Also, I discarded adding a "duration" property because curves can be infinite.

**Comment by Jjagg** [comment]
Fixed in #7387.
